packages/cli/src/lib/inputHelpers.ts

```diff
@@ -15,38 +15,164 @@
+import {ValidateFunction} from './Prompt';
+import {enUS as messages} from './strings';
+import {domainToASCII} from 'url';
 
-const MIN_KEY_PASSWORD_LENGTH = 6;
+/**
+ * A {@link ValidateFunction} that receives a {@link string} as input and resolves to a
+ * {@link Color} when successful.
+ * @param {string} color a string to be converted to a {@link Color}.
+ * @returns {Result<Color, Error} a results that resolves to a {@link Color} on success or
+ * {@link Error} on failure.
+ */
+export function validateColor(color: string): Result<Color, Error> {
+  try {
+    return Result.ok(new Color(color));
+  } catch (_) {
+    return Result.error(new Error(messages.errorInvalidColor(color)));
+  }
+};
+
+/**
+ * A {@link ValidateFunction} that receives a {@link string} as input and resolves to a
+ * {@link URL} when successful. If the string is empty, the validation fails and the the
+ * {@link Result} returned by the function is an {@link Error}.
+ * @param {string} url a string to be converted to a {@link URL}.
+ * @returns {Result<URL, Error} a results that resolves to a {@link URL} on success or
+ * {@link Error} on failure.
+ */
+export function validateUrl(url: string): Result<URL, Error> {
+  if (isWebUri(url) === undefined) {
+    return Result.error(new Error(messages.errorInvalidUrl(url)));
+  }
 
-export async function validateKeyPassword(input: string): Promise<boolean> {
-  if (input.trim().length < MIN_KEY_PASSWORD_LENGTH) {
-    throw new Error(`Password must be at least ${MIN_KEY_PASSWORD_LENGTH} characters long`);
+  try {
+    return Result.ok(new URL(url));
+  } catch (e) {
+    return Result.error(new Error(messages.errorInvalidUrl(url)));
   }
-  return true;
 }
 
-export async function notEmpty(input: string, fieldName: string): Promise<boolean> {
-  const error = util.validateNotEmpty(input, fieldName);
-  if (error) {
-    throw new Error(error);
+/**
+ * A {@link ValidateFunction} that receives a {@link string} as input and resolves to a
+ * {@link URL} or {@link null} when successful. If the string is empty, the validation succeeds
+ * and the {@link Result} returned by the function {@link null}. Non-empty strings are validated
+ * and a conversion is attempted.
+ * @param {string} url a string to be converted to a {@link URL}.
+ * @returns {Result<URL, Error} a results that resolves to a {@link URL} on success or
+ * {@link Error} on failure.
+ */
+export function validateOptionalUrl(input: string): Result<URL | null, Error> {
+  const url = input.trim();
+  if (url.length === 0) {
+    return Result.ok(null);
   }
-  return true;
+
+  return validateUrl(url);
 }
 
-export async function validateColor(color: string): Promise<boolean> {
+/**
+ * Creates a {@link ValidateFunction<string>} that checks the input {@link string} against the
+ * constraints provided as parameters.
+ * @param {number?} minLength optional minimum length for the string.
+ * @param {number?} maxLength optional maximum length for the string.
```

Comment:
nit: duplicated line

Reply:
those are actually different. One for `minLength` and the other for `maxLength`. Tried making the sentence shorter and highlighting where it's different.